### PR TITLE
Enable watchdog for ArduinoCore-mbed based boards

### PR DESCRIPTION
**Pull request by aentinger**

### CODE REVIEW

File: src/utility/watchdog/Watchdog.cpp [hunk 1]
```diff
@@ -56,3 +73,31 @@ void wifi_nina_feed_watchdog()
   samd_watchdog_reset();
 }
 #endif /* ARDUINO_ARCH_SAMD */
+
+#ifdef ARDUINO_ARCH_MBED
+void mbed_watchdog_enable()
+{
+  watchdog_config_t cfg;
+#if defined(ARDUINO_PORTENTA_H7_M7) || defined(ARDUINO_PORTENTA_H7_M)
```
Comment (facchinm) [inline src/utility/watchdog/Watchdog.cpp:81]: `ARDUINO_PORTENTA_H7_M` should be `ARDUINO_PORTENTA_H7_M4` :wink:

File: src/utility/watchdog/Watchdog.h [hunk 2]
```diff
@@ -18,21 +18,6 @@
...
- ******************************************************************************/
-
-#include <AIoTC_Config.h>
-
-/******************************************************************************
- * PREPROCESSOR SECTION
- ******************************************************************************/
-
-#ifdef ARDUINO_ARCH_SAMD
-#  include <Adafruit_SleepyDog.h>
```
Comment (facchinm) [inline src/utility/watchdog/Watchdog.h:32]: Is sleepydog included somewhere else?

**Review by facchinm — CHANGES_REQUESTED** [review]

File: src/utility/watchdog/Watchdog.h (hunk 2 above)
Comment (aentinger) [reply src/utility/watchdog/Watchdog.h:32]: No, that's the only spot. But this isn't a change particular to this PR, I simply moved the inclusion from .h to .cpp.

File: src/utility/watchdog/Watchdog.cpp (hunk 1 above)
Comment (aentinger) [reply src/utility/watchdog/Watchdog.cpp:81]: Good catch!

**Pull request merged**